File: initial/src/test/java/com/example/restservice/RestServiceApplicationTests.java
package com.example.restservice;

// JUnit 5のテスト用アノテーション
import org.junit.jupiter.api.Test;

// テスト時にSpringアプリケーション全体を起動するアノテーション
import org.springframework.boot.test.context.SpringBootTest;

/*
 * RestServiceApplicationTests.java - アプリケーション起動テスト
 *
 * 【解説】
 * このテストクラスは、Spring Bootアプリケーションが正常に起動できるかを確認します。
 * contextLoads() メソッドは中身が空ですが、@SpringBootTest により
 * Springのアプリケーションコンテキスト（設定・Bean全体）が正しくロードされることを検証します。
 *
 * もしアプリケーションの設定にエラーがあれば、このテストが失敗します。
 * 「スモークテスト」とも呼ばれ、最低限の動作確認を行うテストです。
 */
@SpringBootTest
public class RestServiceApplicationTests {

  // アプリケーションコンテキストが正常にロードされるかを確認するテスト
  // メソッド内が空 = 例外が発生しなければテスト成功
  @Test
  public void contextLoads() {
  }

}

File: initial-kotlin/src/test/kotlin/com/example/restservice/RestServiceApplicationTests.kt
package com.example.restservice

// JUnit 5のテスト用アノテーション
import org.junit.jupiter.api.Test
// テスト時にSpringアプリケーション全体を起動するアノテーション
import org.springframework.boot.test.context.SpringBootTest

/*
 * RestServiceApplicationTests.kt - アプリケーション起動テスト（Kotlin版）
 *
 * 【解説】
 * Spring Bootアプリケーションが正常に起動できるかを確認するテストです。
 * contextLoads() の中身は空ですが、@SpringBootTest によりアプリケーション全体の
 * コンテキスト（設定やBean）が正しくロードされることを検証します。
 * 設定にエラーがあれば自動的にテストが失敗します。
 */
@SpringBootTest
class RestServiceApplicationTests {

  // アプリケーションコンテキストのロードテスト
  // 中身が空 = 例外が発生しなければ成功
  @Test
  fun contextLoads() {
  }

}
